tests/unit/stream_alert_cli/test_config.py

```diff
@@ -14,67 +14,73 @@
 limitations under the License.
 """
 # pylint: disable=protected-access
-from contextlib import nested
 import json
 
 from mock import Mock, patch
 from nose.tools import assert_equal, assert_true
+from pyfakefs import fake_filesystem_unittest
 
 from stream_alert_cli.config import CLIConfig
-from tests.unit.helpers.base import basic_streamalert_config, mock_open
+from tests.unit.helpers.base import basic_streamalert_config
 
 
 class TestCLIConfig(object):
     """Test class for CLIConfig"""
+    # pylint: disable=no-self-use
 
     def __init__(self):
-        self.mocked_opens = None
+        self.fs_patcher = None
 
     def setup(self):
         """Setup before each method"""
         config_data = basic_streamalert_config()
-        self.mocked_opens = [
-            mock_open('conf/global.json', json.dumps(config_data['global'])),
-            mock_open('conf/lambda.json', json.dumps(config_data['lambda'])),
-            mock_open('conf/clusters/prod.json', json.dumps(config_data['clusters']['prod']))
-        ]
+
+        self.fs_patcher = fake_filesystem_unittest.Patcher()
```

Comment:
Question: Could `fake_filesystem_unittest.TestCase` be subclassed here to get all of this functionality for free?

Reply:
You can, but it will convert the test case into a `unittest` test case, as `fake_filesystem_unittest.TestCase` inherits from `unittest.TestClass` (https://github.com/jmcgeheeiv/pyfakefs/wiki/Automatically-find-and-patch-file-functions-and-modules#unit-test-class).

Which `nose` can run, but we will have to change the setup method from `setup(self)` to `setUp(self)`. It may end up being more confusing.

Reply:
Good point - let's leave it!